railties/lib/rails/api/generator.rb

```diff
@@ -0,0 +1,47 @@
+require "sdoc"
+
+class RDoc::Generator::API < RDoc::Generator::SDoc # :nodoc:
+  RDoc::RDoc.add_generator self
+
+  alias :original_generate_class_tree_level :generate_class_tree_level
+
+  def generate_class_tree_level(classes, visited = {})
+    # If the method is called for the first time, we just
+    # extract the core classes from the list.
+    first_call = visited == {}
```

Comment:
On second thought, why do we need the `first_call` check at all? I don't understand the intent of the comment either.